pkg/sql/schemachanger/scexec/scmutationexec/index.go

```diff
@@ -309,3 +293,74 @@ func (m *visitor) SetIndexName(ctx context.Context, op scop.SetIndexName) error
 	index.IndexDesc().Name = op.Name
 	return nil
 }
+
+func (m *visitor) AddColumnToIndex(ctx context.Context, op scop.AddColumnToIndex) error {
+	tbl, err := m.checkOutTable(ctx, op.TableID)
+	if err != nil {
+		return err
+	}
+	index, err := tbl.FindIndexWithID(op.IndexID)
+	if err != nil {
+		return err
+	}
+	column, err := tbl.FindColumnWithID(op.ColumnID)
+	if err != nil {
+		return err
+	}
+	// Deal with the fact that we allow the columns to be unordered in how
+	// we add them. We could add a rule to make sure we add the columns in
+	// order, but we'd need a way to express successor or something in rel
+	// rules to add those dependencies efficiently. Instead, we just don't
+	// and sort here.
+	id := index.IndexDesc()
```

Comment:
nit: I think `idx` is a better name bc when I first read this block, my mind constantly mistakenly think it as some sort of ID